src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -147,34 +150,40 @@ public UnfilteredRowIterator queryStorage(PrimaryKey key, ReadExecutionControlle
     }
 
     /**
-     * Build a {@link KeyRangeIterator.Builder} from the given list of expressions by applying given operation (AND).
-     * Building of such builder involves index search, results of which are persisted in the internal resources list
-     *
-     * @param expressions The expressions to build range iterator from (expressions with not results are ignored).
-     *
-     * @return range iterator builder based on given expressions and operation type.
+     * Build a {@link KeyRangeIterator.Builder} from the given list of {@link Expression}s.
+     * <p>
+     * This is achieved by creating an on-disk view of the query that maps the expressions to
+     * the {@link SSTableIndex}s that will satisfy the expression.
+     * <p>
+     * Each (expression sstable indexes) pair is then passed to
+     * {@link IndexSearchResultIterator#build(Expression, Collection, AbstractBounds, QueryContext)}
+     * to search the in-memory index associated with the expression and the sstable indexes, the results of
+     * which are unioned and returned. The results from each expression are added to a
+     * {@link KeyRangeIntersectionIterator} and returned.
```

Comment:
super nit: Feels like the last sentence and the last clause of the previous one are saying the same thing :p

Reply:
Not really but I have made it clearer. The results from each call to `IndexSearchResultIterator.build` are added to a `KeyRangeIntersectionIterator` and returned.